python/compiler/txn2mlir.py

```diff
@@ -137,49 +139,107 @@ def device_body():
                     else:
                         global_data.append(None)
 
-                @runtime_sequence()
-                def sequence():
-                    for op, payload in zip(operations, global_data):
-                        if op[0] == 0x00:
-                            addr = op[1]
-                            value = op[2]
-                            npu_write32(addr, value)
-                        elif op[0] == 0x01:
-                            addr = op[1]
-                            d = memref.get_global(
-                                payload.type_.value, payload.sym_name.value
-                            )
-                            npu_blockwrite(addr, d)
-                        elif op[0] == 0x03:
-                            addr = op[1]
-                            value = op[2]
-                            mask = op[3]
-                            npu_maskwrite32(addr, value, mask)
-                        else:
-                            raise Exception(f"Unhandled op: {op:#x}")
+                if mlir_ctrl_pkt:
```

Comment:
This is a nice quick way to get a proof of concept but I think npu op to control packet conversion should be an mlir pass in C++ if it is going to be supported in mlir-aie.  Converting runtime sequence mlir to a binary blob then to control packet mlir is a bit hacky.

Reply:
Totally agreed. I'm thinking that perhaps ideally there should be an mlir pass that converts write32/blockwrite/maskedwrite32 ops into control packet ops, and another pass that converts control packet ops to dma_memcpy_nd ops represeting shim dma issuing control packet writes to tiles. Let me create an issue to extend this discussion.

Reply:
Sounds good. This txn2mlir.py script is mostly intended to be a disassembler and debugging aid. It is not intended to be load bearing.

Reply:
Issue created: https://github.com/Xilinx/mlir-aie/issues/1729

Reply:
related to https://github.com/Xilinx/mlir-aie/pull/1740